cmd/nlb_list.go

```diff
@@ -0,0 +1,91 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	apiv2 "github.com/exoscale/egoscale/api/v2"
+	"github.com/spf13/cobra"
+)
+
+type nlbListItemOutput struct {
+	ID        string `json:"id"`
+	Name      string `json:"name"`
+	Zone      string `json:"zone"`
+	IPAddress string `json:"ip_address"`
+}
+
+type nlbListOutput []nlbListItemOutput
+
+func (o *nlbListOutput) toJSON()  { outputJSON(o) }
+func (o *nlbListOutput) toText()  { outputText(o) }
+func (o *nlbListOutput) toTable() { outputTable(o) }
+
+var nlbListCmd = &cobra.Command{
+	Use:   "list",
+	Short: "List Network Load Balancers",
+	Long: fmt.Sprintf(`This command lists Network Load Balancers.
+
+Supported output template annotations: %s`,
+		strings.Join(outputterTemplateAnnotations(&nlbListItemOutput{}), ", ")),
+	Aliases: gListAlias,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		zone, err := cmd.Flags().GetString("zone")
+		if err != nil {
+			return err
+		}
+		zone = strings.ToLower(zone)
+
+		return output(listNLBs(zone), nil)
+	},
+}
+
+func listNLBs(zone string) outputter {
+	var nlbZones []string
+
+	if zone != "" {
+		nlbZones = []string{zone}
+	} else {
+		nlbZones = zones
+	}
+
+	out := make(nlbListOutput, 0)
+	res := make(chan nlbListItemOutput)
+	defer close(res)
+
+	go func() {
+		for nlb := range res {
+			out = append(out, nlb)
+		}
+	}()
```

Comment:
IMO this go routine is not useful in this case, can be appended directly in the result loop.

Reply:
Tempting, but after a quick search it looks like it's not a good idea to append to a slice from multiple goroutines:

https://groups.google.com/g/golang-nuts/c/SRtSTar853g?pli=1
https://medium.com/@cep21/gos-append-is-not-always-thread-safe-a3034db7975
http://dnaeon.github.io/concurrent-maps-and-slices-in-go/

Alternatively, we could wrap the slice in a new type that includes a `sync.Mutex` for locking, but isn't it overkill?

Reply:
After looking more deeply, didn't see your goroutine in `forEachZone `. Well let's keep like that 😉

Reply:
>Alternatively, we could wrap the slice in a new type that includes a sync.Mutex for locking, but isn't it overkill?

Yes let's keep the go routine with the chan in this case